### ISSUE TITLE
Support for AlphaFold database

### ISSUE THREAD

**Issue by sirius777coder** [biotite-dev/biotite#438]
Hi, 

I think Biotite is a really useful tool when I deal with a lot of biological data. Do your team has any plan to integrate AlphaFold database into the biotite API?

**Comment by padix-key** [comment]
Hi, currently it is not planned to add a direct interface to the *AlphaFold Protein Structure Database* (https://alphafold.ebi.ac.uk/). To my knowledge this database has no documentation for a REST API to search AlphaFold structures and fetch them. The RCSB PDB started to intgrate AlphaFold structures itself (https://www.rcsb.org/docs/general-help/computed-structure-models-and-rcsborg), so probably download of AlphaFold structures will be managed via `biotite.database.rcsb` in the foreseeable future.

**Comment by sirius777coder** [comment]
OK, look forward to seeing this!

**Comment by BradyAJohnston** [comment]
AlphaFoldDB officially released docs and better support for their API. I was going to have a go at it, but I don't really know my way around APIs much at all. Thought I would share the links to potentially have it integrated into biotite. 

https://www.ebi.ac.uk/about/news/updates-from-data-resources/alphafold-database-ux-update/

https://alphafold.ebi.ac.uk/api-docs

**Comment by dacarlin** [comment]
I would be happy to contribute an interface to the AlphaFold DB, I am familiar with the APIs. Is this something the project would welcome? I would be a first time contributor to biotite (long time user)

**Comment by padix-key** [comment]
Yes, indeed! In the moment there is already a contributor that was working in this in #465. However, the last commit is multiple months ago, so I expect the contributor cancelled the project. I would like to give the author another week to respond. Otherwise I would close his PR and give way to you to work on this feature.

I would imagine an interface that is similar to `biotite.database.pdb` (at least for `fetch()`). Although for license reasons you cannot simply copy the code from #465, you can still take advantage of the discussion.

**Comment by dacarlin** [comment]
That all sounds good to me 👍 Agree that the interface for `biotite.database.pdb` is a perfect model. How about I check back next week for an updated status?

**Comment by padix-key** [comment]
There was no response, yet, so I closed #465. Hence you can start if you like.

**Comment by dacarlin** [comment]
That is wonderful, thanks for the update! I will plan on working on this over the next week or so. Will report back once I have something to showSent from my iPhoneOn Jul 19, 2023, at 9:24 AM, Patrick Kunzmann ***@***.***> wrote:﻿
Now @jonfunk21 also approved, that you may take on the issue.

**Comment by dacarlin** [comment]
I’ve implemented fetching and included a few basic tests so far (see here https://github.com/dacarlin/biotite/blob/add-alphafold-db/tests/database/test_alphafold.py).

Shall I open a WIP merge request as I wrap up the changes and get the code ready for your comments? Currently, I have my changes in a branch that is in a recent fork

**Comment by padix-key** [comment]
Looks already quite good, feel free to open a PR. I wonder if the tests pass, since there are two issues, if I am not wrong.

`content = file_response.text` should not work for `bcif` because it is a binary format.

´from .check import assert_valid_response´ should give an error because there is no `check.py`module in this subpackage.

**Comment by dacarlin** [comment]
Yes, thanks! You are definitely correct, I ran into issues installing and running the tests which I think I may need a bit of help debugging. 

Briefly, I created a new Conda environement using the `environment.yml` but I still get some errors from Cython that are preventing me from installing with `pip install .` and `pip install -e .`. 

I put the error log in this gist https://gist.github.com/dacarlin/bac823ef62aef6d3df7261ad8dd6d76a. If you have any suggestions for how to get the install working I can finish up the tests and make the PR!

**Comment by padix-key** [comment]
I suppose you probably have the brand new Cython 3.0 installed? I just saw that the `environment.yml` does not restrict the upper bound of the Cython version so you probably have installed version 3.0, that introduced some breaking language changes. It quite relieves me to hear, that only a few lines of code seem to require fixing. I will create an issue for the changes introduced by Cython 3.0.

**Comment by dacarlin** [comment]
Glad that it's an easy fix. This is the output of `conda env export --from-history` in case it helps troubleshoot 

```
name: biotite-dev
channels:
  - defaults
dependencies:
  - sphinx-gallery=0.11.1
  - tantan
  - matplotlib[version='>=3.3']
  - networkx[version='>=2.0']
  - muscle=3
  - setuptools[version='>=30.0']
  - mafft
  - sphinx[version='>=5.0']
  - numpy[version='>=1.15']
  - numpydoc[version='>=0.8']
  - requests[version='>=2.12']
  - sra-tools
  - dssp
  - wheel[version='>=0.30']
  - autodock-vina
  - scipy[version='>=1.8.0']
  - pytest[version='>=5.2']
  - scikit-learn[version='>=0.18']
  - pydot[version='>=1.4']
  - mdtraj[version='>=1.9.3']
  - msgpack-python[version='>=0.5.6']
  - clustalo
  - cython[version='>=0.29']
  - viennarna[version='>=2.5.0']
  - python=3.9
  - sphinxcontrib-bibtex[version='>=2.3']
  - pip[version='>=10.0']
  - cmake
```

Meanwhile, I'll create a PR for this feature branch 😄

**Comment by padix-key** [comment]
Implemented by #492
